### PR TITLE
feat(mojaloop/3268): refactored config library and special char support in mongodb config

### PR DESCRIPTION
**Pull request by vijayg10**
feat([mojaloop/3268](https://github.com/mojaloop/project/issues/3268)): refactored config library and special char support in mongodb config

### CODE REVIEW

File: config/test.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (mdebarros) [inline config/test.json]: Why do we need this, as it looks identical to `default.json`?

I prefer not to include "environment" configs as part of our repos.

File: package.json [hunk 2]
```diff
@@ -23,11 +25,11 @@
...
-    "npm-run-all": "^4.1.5",
-    "standard": "^16.0.4",
     "npm-audit-resolver": "^2.3.1",
     "npm-check-updates": "^11.8.5",
-    "pre-commit": "1.2.2"
+    "npm-run-all": "^4.1.5",
+    "pre-commit": "1.2.2",
+    "standard": "^16.0.4"
   },
   "scripts": {
```
Comment (mdebarros) [inline package.json:34]: Interesting.

We don't have the usual "release" or even "commit" lint scripts.

I think we may need to add some stories to fix that up.

Perhaps we can group the nodejs upgrade with that story. :/

File: src/lib/config.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      format: String,
+      default: 'topic-event',
+      env: 'KAFKA_TOPIC_EVENT',
+    },
+  },
+})
+
+// Load environment dependent configuration
+const env = ConvictConfig.get('env')
+const configFile = process.env.CONFIG_FILE || path.join(__dirname, `../../config/${env}.json`)
```
Comment (mdebarros) [inline src/lib/config.js]:
```suggestion
const configFile = process.env.CONFIG_FILE || path.join(process.cwd(), `./config/${env}.json`)
```
☝️ I prefer this approach as its cleaner. Let me know your thoughts.

File: src/services/event-processor.service.js [hunk 4]
```diff
@@ -48,6 +49,8 @@ class EventProcessorService {
         return
       }
 
+      console.log(`Processing eventId: ${msg.id}`)
```
Comment (mdebarros) [inline src/services/event-processor.service.js:52]: Console.log?

File: src/services/mongo-db.service.js [hunk 5]
```diff
@@ -13,18 +14,18 @@ class MongoDBService {
     try {
       await mongo.connect()
 
-      const res = await mongo.db(process.env.MONGO_DB_DBNAME).command({ ping: 1 })
+      const res = await mongo.db().command({ ping: 1 })
```
Comment (mdebarros) [inline src/services/mongo-db.service.js:17]: Nice, this is so much cleaner.

**Review by mdebarros — COMMENTED** [review]
LGTM, just some minor comments for you to address.

File: package.json (hunk 2 above)
Comment (vijayg10) [reply package.json:34]: Sure and also we need to standardise this with mojaloop repos like CICD, package.json...etc

File: src/services/event-processor.service.js (hunk 4 above)
Comment (vijayg10) [reply src/services/event-processor.service.js:52]: There is no logger in this service. console.log statements everywhere. May be we need a story to streamline that.

File: config/test.json (hunk 1 above)
Comment (vijayg10) [reply config/test.json]: I removed env based config files logic as we discussed. Thanks.

Comment (mdebarros) [reply config/test.json]: @vijayg10 do we still need this file?

Comment (vijayg10) [reply config/test.json]: I missed to delete the file, my bad. Deleted now.

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**